### DISCUSSION TITLE
Wrong mapping of go packageName in case endpoint contains "api/v4" and packageName "api/"

### DISCUSSION THREAD

**Discussion by CodeColumbus** [renovatebot/renovate#27595] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

In case a go package has "api/" as part of the name and the endpoint contains "api/v4" (like it is with a self-hosted GitLab instance), packageName will be mapped in a wrong way. We are facing 404 response from GitLab Tags API because there is no matching project as "api/" is not part of the package name anymore.

Please take a look at https://github.com/CodeColumbus/renovate/commit/b4cc25b69b1d35c075a716723c84da27f944df86
I created a test to reproduce and already tried to modify endpointPrefix regex to fix this issue:
Endpoint: https://my.custom.domain/api/v4
PackageName: golang/subgroup-api/myrepo
Result: golang/subgroup-myrepo :(

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by CodeColumbus** [comment]
sry the link to the commit is not working anymore. 
here is the test I created: https://github.com/CodeColumbus/renovate-fork/pull/1/files

**Reply by rarkins** [reply]
This test represents a test which should pass, but which currently fails?

**Reply by CodeColumbus** [reply]
Hi!
yes. 
The test fails because "api/" was removed from packageName.
I would expect an unchanged packageName"golang/subgroup-api/myrepo" as result. But it is "golang/subgroup-myrepo"

It seems like this issue was introduced here: 
https://github.com/renovatebot/renovate/pull/26477/files
Because after I removed this line: https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/go/base.ts#L168
packageName will not be touched anymore and the test passes.

**Reply by CodeColumbus** [reply]
maybe it is confusing that I used "golang/". I am not sure what that pull request tried to achieve.
This issue has nothing to do with "golang/". I changed it in
https://github.com/CodeColumbus/renovate-fork/pull/1/files from "golang/" to "group/" and the test is still failing.
because packageName is changed to "group/subgroup-myrepo", but should be "group/subgroup-api/myrepo".

**Reply by rarkins** [reply]
I don't understand what you're saying above but @viceice wrote the PR and may

**Reply by viceice** [reply]
🤔 it should only remove `api/v4` if it ends with that. 🤔

maybe the used regex capture group number is wrong.

btw: I haven't written that PR. 😉

**Reply by rarkins** [reply]
True, it was @ManuelB

**Reply by ManuelB** [reply]
Sorry for introducing this bug. I hope we can fix it soon.

**Reply by viceice** [reply]
@ManuelB Do you have an idea how to fix this?

**Reply by ManuelB** [reply]
I would guess that I can just limit the regex to the end of the string.

**Reply by CodeColumbus** [reply, marked as answer]
Good morning,
I just saw that https://github.com/renovatebot/renovate/pull/27635 is already merged 👌 
Thank you for resolving this issue so quickly👍
